#!/usr/bin/env python

PACKAGE = "robot_micro"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
gen.add("S_kp", double_t, 0, "S_KP", 0.3, 0, 2)
gen.add("S_kd", double_t, 0, "S_KD", 0.02, 0, 2)
gen.add("S_ki", double_t, 0, "S_KI", 0.1, 0, 2)
gen.add("S_wu", double_t, 0, "S_wu", 1.0, 0, 2)

gen.add("P_kp", double_t, 0, "S_KP", 0.3, 0, 2)
gen.add("P_kd", double_t, 0, "S_KD", 0.02, 0, 2)
gen.add("P_ki", double_t, 0, "S_KI", 0.1, 0, 2)
gen.add("P_wu", double_t, 0, "S_wu", 1.0, 0, 2)

gen.add("scaling_joy", double_t, 0, "scaling_joy", 1.0, 0, 5)

exit(gen.generate(PACKAGE, "robot_micro", "controller"))